File: server/src/main/java/pt/iade/ei/thinktoilet/models/requests/RegisterRequest.java
package pt.iade.ei.thinktoilet.models.requests;

import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class RegisterRequest {
    @Size(min = 6, max = 50)
    private String name;
    @Size(max = 100)
    private String email;
    @Size(min = 6)
    private String password;
    private String iconId;
    private LocalDate birthDate;
}

File: app/app/src/main/java/pt/iade/ei/thinktoilet/models/requests/RegisterRequest.kt
package pt.iade.ei.thinktoilet.models.requests

import com.google.gson.annotations.SerializedName

data class RegisterRequest(
    @SerializedName("name") val name: String,
    @SerializedName("email") val email: String,
    @SerializedName("password") val password: String,
    @SerializedName("iconId") val iconId: String?,
    @SerializedName("birthDate") val birthDate: String?
)
